harvester_e2e_tests/integration/test_vm.py

```diff
@@ -0,0 +1,327 @@
+    vm_names = [f"migrate-1-{unique_name}", f"migrate-2-{unique_name}"]
+
+    vm_spec = api_client.vms.Spec(1, 1)
+    vm_spec.add_image('disk-0', focal_image['id'])
+    vm_data = []
+    for vm_name in vm_names:
+        code, data = api_client.vms.create(vm_name, vm_spec)
+        assert 201 == code, (
+            f"Failed to create VM {vm_name} with error: {code}, {data}"
+        )
+        vm_data.append(data)
+
+    for vm_name in vm_names:
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        code, data = None, None
+        while endtime > datetime.now():
+            code, data = api_client.vms.get(vm_name)
+            if data.get('status', {}).get('ready', False):
+                code, data = api_client.vms.get_status(vm_name)
+                if data['status']['conditions'][-1]['status'] == 'True':
+                    break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Can't find VM {vm_name} with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+    vmi_data = []
+    for vm_name in vm_names:
+        code, data = api_client.vms.get_status(vm_name)
+        assert 200 == code, (
+            f"Failed to get VM {vm_name} with error: {code}, {data}"
+        )
+        vmi_data.append(data)
+
+    vm_src_dst_hosts = {}  # {vm_name: [src_host, dst_host]}
+    for vm_name, vm_datum in zip(vm_names, vmi_data):
+        src_host = vm_datum['status']['nodeName']
+        dst_host = next(n for n in available_node_names if n != src_host)
+        vm_src_dst_hosts[vm_name] = [src_host, dst_host]
+
+        code, data = api_client.vms.migrate(vm_name, dst_host)
+        assert code == 204, (
+            f"Can't migrate VM {vm_name} from host {src_host} to {dst_host}",
+            f"Got error: {code}, {data}"
+        )
+
+    for vm_name, src_dst_hosts in vm_src_dst_hosts.items():
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        code, data = None, None
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(vm_name)
+            if data.get('status', {}).get('migrationState', {}).get('completed', False):
+                assert src_dst_hosts[1] == data['status']['nodeName'], (
+                    f"Failed to migrate VM {vm_name} \
+                        from {src_dst_hosts[0]} to {src_dst_hosts[1]}"
+                )
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"The migration of VM {vm_name} is not completed with {wait_timeout} timed out"
+                f"Got error: {code}, {data}"
+            )
+
+    # teardown
+    for vm_name in vm_names:
+        api_client.vms.delete(vm_name)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        code, data = None, None
+        while endtime > datetime.now():
+            code, data = api_client.vms.get(vm_name)
```

Comment:
I had encountered that VM definition has been removed but VMI still exists, so I would suggest to use `get_status` to make sure VMI been removed. (to avoid delete volumes being deny.)